.github/actions/setup_test_environ/action.yml

```diff
@@ -0,0 +1,63 @@
+name: "Setup test environment"
+inputs:
+  ARTIFACT_RUN_ID:
+    type: string
+    required: true
+  AMDGPU_FAMILIES:
+    type: string
+  OUTPUT_ARTIFACTS_DIR:
+    type: string
+    required: true
+  FETCH_ARTIFACT_ARGS:
+    type: string
+  VENV_DIR:
+    type: string
+    required: true
+runs:
+  using: "composite"
+  steps:
+    - name: Install the AWS tool
+      shell: bash
+      run: ./dockerfiles/cpubuilder/install_awscli.sh
+
+    - name: "Setting up Python"
+      id: setup_python
+      uses: actions/setup-python@42375524e23c412d93fb67b49958b491fce71c38 # v5.4.0
+      with:
+        python-version: 3.11
+
+    - name: Create Python venv
+      shell: bash
+      run: |
+        python -m venv ${VENV_DIR}
+        source ${VENV_DIR}/bin/activate
+        pip install -r requirements-test.txt
+      env:
+        VENV_DIR: ${{ inputs.VENV_DIR }}
+
+    - name: Download and Unpack Artifacts
+      shell: bash
+      run: |
+        source ${VENV_DIR}/bin/activate
+
+        # TODO(sosa): We should infer this from the runner. For Libs and Above it should be based on the
+        # runner, everything else is generic. Use just generic since this is all core/below.
+        echo "Making ${OUTPUT_ARTIFACTS_DIR} directory"
+        mkdir -p "${OUTPUT_ARTIFACTS_DIR}"
+
+        echo "Downloading artifacts"
+        python ./build_tools/fetch_artifacts.py \
+          --run-id=${ARTIFACT_RUN_ID} \
+          --target=${AMDGPU_FAMILIES} \
+          --build-dir=${OUTPUT_ARTIFACTS_DIR} \
+          ${FETCH_ARTIFACT_ARGS}
+
+        echo "Unpacking artifacts"
+        mkdir ${OUTPUT_ARTIFACTS_DIR}/output_dir
+        python ./build_tools/fileset_tool.py artifact-flatten ${OUTPUT_ARTIFACTS_DIR}/*.tar.xz -o ${OUTPUT_ARTIFACTS_DIR}/output_dir --verbose
+      env:
```

Comment:
I don't know what is the best practices here but as the environment variables are used in the `run` _above_ I would prefer swap the order of `env` and `run`.